caas/kubernetes/provider/k8s.go

```diff
@@ -468,17 +481,19 @@ func (k *kubernetesClient) EnsureService(
 		return errors.Annotatef(err, "parsing unit spec for %s", appName)
 	}
 
-	// Now add in any storage requested.
-	if err := k.configureStorage(unitSpec, appName, params); err != nil {
-		return errors.Annotatef(err, "configuring storage for %s", appName)
-	}
-
 	// Add a deployment controller configured to create the specified number of units/pods.
 	numPods := int32(numUnits)
-	if err := k.configureDeployment(appName, unitSpec, params.PodSpec.Containers, &numPods); err != nil {
-		return errors.Annotate(err, "creating or updating deployment controller")
+	if len(params.Filesystems) > 0 {
+		if err := k.configureStatefulSet(appName, unitSpec, params.PodSpec.Containers, &numPods, params.Filesystems); err != nil {
+			return errors.Annotate(err, "creating or updating stateful set")
```

Comment:
is it `the StatefulSet` here?

Reply:
yeah, so if the unit requires storage, we use a stateful set
if there's no storage required, we continue to use a deployment controller as it's lighter weight